internal/services/provisionner.go

```diff
@@ -61,34 +61,47 @@ func GenerateResources() error {
 		blackWhiteList = MakeBlackWhitelist(blacklistCM.Data)
 	}
 
-	GenerateProjects(auths, &blackWhiteList)
+	createdproject, deletedprojects, ignoredProjects := GenerateProjects(auths, &blackWhiteList)
+	for _, project := range ignoredProjects {
+		utils.Log.Error().Msgf("Cannot find project %s in whitelist", project.Namespace())
+	}
+	for _, project := range deletedprojects {
+		utils.Log.Info().Msgf("delete project %s in blacklist", project.Namespace())
+		deleteProject(project)
+	}
+	// now that the project is well categorized we know that a project cannot be at the same  time to be deleted and to be generated
+	for _, project := range createdproject {
+		utils.Log.Info().Msgf("Project %s is whitelisted", project.Namespace())
+		generateProject(project)
+	}
 	return nil
 }
 
 // A loop wrapper for generateProject
 // splitted for unit test !
-func GenerateProjects(context []*types.Project, blackWhiteList *types.BlackWhitelist) {
+func GenerateProjects(context []*types.Project, blackWhiteList *types.BlackWhitelist) ([]*types.Project, []*types.Project, []*types.Project) {
```

Comment:
we create new project slices, so it would have been better to not point to []* but let's keep this for now